src/scenes/StageScene.js

```diff
@@ -78,7 +83,11 @@ export default class StageScene extends Phaser.Scene {
       // create opponents
       Object.keys(this.roomInfo.players).forEach((playerId) => {
         if (playerId !== this.socket.id) {
-          this.opponents[playerId] = this.createPlayer(this.roomInfo.players[playerId].spriteKey);
+          this.opponents[playerId] = this.createPlayer(this.roomInfo.players[playerId].spriteKey, this.roomInfo.players[playerId].username);
+          this.opponentNameText = this.add.text(this.opponents[playerId].x, this.opponents[playerId].y - 16, this.roomInfo.players[playerId].username, { 
```

Comment:
this line is storing all opponents' text box on the same property of the scene, so this.opponentNameText will only point to the last text box of the last opponent's username